### DISCUSSION TITLE
Min/Max Zoom, remove/constrain certain movement of models

### DISCUSSION THREAD

**Discussion by lolo2k** [mkkellogg/GaussianSplats3D#326] [Ideas]
Hi, We would love to have a max zoom in, max zoom out value. And in a similar vain, no horizontal or vertical translation (and/or other movement of models like vertical rotation). Then object can be displayed without unnessasry angles. Thanks

**Comment by mkkellogg** [comment]
You can technically achieve this right now, you just have to access the controls owned by the viewer. The viewer has two sets of controls: `Viewer.perspectiveControls` and `Viewer.orthographicControls`. You'd need to set the same parameters for both:

```javascript

const setControlParameters(controls) = (controls, isOrthographic) => {
  if (isOrthographic) {
    controls.minZoom = 1.0;
    controls.maxZoom = 10.0;
  } else {
    controls.minDistance = 3.0;
    controls.maxDistance = 100.0;
  }
  controls.enablePan = false; // horizontal and vertical translation
};

const viewer = new GaussianSplats3D.Viewer();
viewer.addSplatScene('<path to scene>')
.then(() => {
  setControlParameters(viewer.perspectiveControls, false);
  setControlParameters(viewer.orthographicControls, true);
});


```

**Reply by lolo2k** [reply]
Amazing, thank you for this insight.

**Reply by lolo2k** [reply]
Can I also bother you to tell me the function for rotation? We would like to max the rotation values as well. 
Or can I simply ask my developer to combine the previous code with the 'rotation' reference?
Thank you

**Reply by mkkellogg** [reply]
Yeah there are built in limitations for horizontal and vertical rotation as well:

```javascript
const setControlParameters(controls) = (controls, isOrthographic) => {
  if (isOrthographic) {
    controls.minZoom = 1.0;
    controls.maxZoom = 10.0;
  } else {
    controls.minDistance = 3.0;
    controls.maxDistance = 100.0;
  }

  // Vertical rotation
  controls.minPolarAngle = 0;
  controls.maxPolarAngle = Math.PI;

  // Horizontal rotation
  controls.minAzimuthAngle = - Infinity;
  controls.maxAzimuthAngle = Infinity;

  controls.enablePan = false; // horizontal and vertical translation
};
```

**Reply by lolo2k** [reply]
Amazing. Great stuff. 
Thank you

**Reply by ramaramx** [reply]
@mkkellogg hi, can i ask.
can enablePan can i custom enable pan for example: horizontaly only ??
